### ISSUE TITLE
GroupBy multiple columns is adding unnecessary space

### ISSUE THREAD

**Issue by vamshiqapita** [ag-grid/ag-grid#5722]
I am submitting **bug report**.

On upgrading to **28.2**  from **27.3** version of **ag-grid**, I am seeing additional space around first column when performing _multuple groupBy_ actions.

The action can be replicated on the following **[stackblitz](https://ag-grid.com/react-data-grid/grouping-group-panel/#example-row-group-panel)**.

### Actions

1. Add `groupDisplayType={'multipleColumns'}` to grid
2. Remove **year** groupBy
3. Add **year** groupBy

When we keep doing this, it is kept on adding additional space.

**Initial Grid**
![image](https://user-images.githubusercontent.com/98312465/198025143-99ec1cac-d0c4-443e-a196-d75d87d98f18.png)

**After removing year**
![image](https://user-images.githubusercontent.com/98312465/198025241-2072f11c-ac98-4303-b21c-c8bbb5704bc5.png)

**Comment by YuerLee** [comment]
Maybe duplicate of #5689.

**Comment by StephenCooper** [comment]
I believe the root cause of this will be the same as the duplicate mentioned above. So I have linked this ticket to the AG-7420 jira as well.

We have added this bug to our backlog and we are tracking it with the following reference and description:
AG-7420 Auto group column layout issue caused by async setAutoGroupColumnDef

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

See whether this item will be part of our next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-7420 ) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/

Thanks again for bringing this up with us - we appreciate your time!
